Kickstarter-iOS/Views/Controllers/DiscoveryPageViewController.swift

```diff
@@ -87,26 +90,25 @@ internal final class DiscoveryPageViewController: UITableViewController {
     self.viewModel.inputs.viewDidDisappear(animated: animated)
   }
 
-  internal override func viewDidLayoutSubviews() {
-    super.viewDidLayoutSubviews()
-
-    self.loadingIndicatorView.center = self.tableView.center
-  }
-
   internal override func bindStyles() {
     super.bindStyles()
 
     _ = self
       |> baseTableControllerStyle(estimatedRowHeight: 200.0)
-
-    _ = self.loadingIndicatorView
-      |> baseActivityIndicatorStyle
   }
 
-    internal override func bindViewModel() {
+  internal override func bindViewModel() {
     super.bindViewModel()
 
-    self.loadingIndicatorView.rac.animating = self.viewModel.outputs.projectsAreLoading
+    self.viewModel.outputs.projectsAreLoadingAnimated
+      .observeForUI()
+      .observeValues { [weak self] (isLoading, animated) in
+        if isLoading {
+          self?.performClosureAnimated(true, closure: { self?.refreshControl?.beginRefreshing() })
```

Comment:
Is it possible to name this function more specifically? Something like 
```
func updateRefreshControl(animated: Bool, _ closure: () -> Void)
```

If we remove the `closure` parameter's argument label, it could be called like this:

```
          self?.updateRefreshControl(animated: animated, {
            self?.refreshControl?.beginRefreshing()
          })
```

Which is less verbose.

Reply:
@Scollaco how come the `performClosureAnimated` function is called with `animated: true` when `isLoading` is `true`, but with `animated: animated` when `isLoading` is false? Is this logic something we could put in the view model so it's not hard-coded on the VC side?

Reply:
If the refreshControl is beginning to refresh, the `animate` property has to be `true`, otherwise it will appear on the screen without animation.